var/spack/repos/builtin/packages/hpctoolkit/package.py

```diff
@@ -198,6 +197,12 @@ class Hpctoolkit(AutotoolsPackage, MesonPackage):
     depends_on("hpcviewer", type="run", when="+viewer")
     depends_on("python@3.10:", type=("build", "run"), when="+python")
 
+    with when("target=x86_64:"):
+        depends_on("intel-xed+pic")
+        patch("modern-xed-2024.01.1.patch", when="@2024.01.1")
+        patch("modern-xed-2023.patch", when="@2022.05.15:2023.08")
+        patch("modern-xed-2022.patch", when="@:2022.04.15")
```

Comment:
Can you add a brief comment above the `patch` directives explaining:
1. What is the effect of the patches
2. Why they are needed

? This would help future reader getting information that are in the PR description, without switching context. Also, if these changes are used upstream by `hpctoolkit` already, can we retrieve the patches from a `url`?

Reply:
Comments added.

> Also, if these changes are used upstream by `hpctoolkit` already, can we retrieve the patches from a `url`?

Unfortunately our build system has gone through enough changes on the branches that the new changes can't be directly applied to the released versions. Hence the need for explicit, specially crafted patches. :disappointed: